### PR TITLE
fix(panel): removes dismissed prop on close

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #4969 

## Summary

This fix will remove `dismissed` prop when the user sets the `closed` prop to `true`

### CODE REVIEW

File: src/components/panel/panel.tsx [hunk 1]
```diff
@@ -56,8 +56,14 @@ export class Panel implements ConditionalSlotComponent, InteractiveComponent {
...
+  dismissedHandler(value: boolean): void {
+    this.closed = value;
+    this.calcitePanelDismissedChange.emit();
+  }
+
   @Watch("closed")
-  dismissedHandler(): void {
+  closedHandler(value: boolean): void {
+    this.dismissed = value;
     this.calcitePanelDismissedChange.emit();
```
Comment (driskull) [inline src/components/panel/panel.tsx:67]: Does this still need to be here? Can you make sure the event isnt being fired twice?

File: src/components/panel/panel.tsx [hunk 2]
```diff
@@ -251,8 +275,8 @@ export class Panel implements ConditionalSlotComponent, InteractiveComponent {
   };
 
   dismiss = (): void => {
-    this.dismissed = true;
     this.closed = true;
+    this.closable = false;
```
Comment (driskull) [inline src/components/panel/panel.tsx:279]: why is closable being set here? closable is independent of closing or dismissing right?

**Review by driskull — CHANGES_REQUESTED** [review]
Some questions here

File: src/components/panel/panel.tsx (hunk 1 above)
Comment (anveshmekala) [reply src/components/panel/panel.tsx:67]: It is firing twice. limited it to single emit now.

File: src/components/panel/panel.tsx (hunk 2 above)
Comment (anveshmekala) [reply src/components/panel/panel.tsx:279]: yup, closable is independent of dismissing.

File: src/components/panel/panel.tsx (hunk 1 above)
Comment (driskull) [reply src/components/panel/panel.tsx:66]: Can we move this to the dismissed watcher?

File: src/components/panel/panel.tsx [hunk 3]
```diff
@@ -251,7 +278,6 @@ export class Panel implements ConditionalSlotComponent, InteractiveComponent {
   };
 
   dismiss = (): void => {
```
Comment (driskull) [inline src/components/panel/panel.tsx:280]: nitpick: maybe we should rename this to `close`

File: src/components/panel/panel.tsx [hunk 4]
```diff
@@ -381,19 +407,18 @@ export class Panel implements ConditionalSlotComponent, InteractiveComponent {
...
-      closable || dismissible ? (
-        <calcite-action
-          aria-label={text}
-          icon={ICONS.close}
-          onClick={dismiss}
-          ref={this.setDismissRef}
-          text={text}
-        />
-      ) : null;
+    const dismissibleNode = closable ? (
```
Comment (driskull) [inline src/components/panel/panel.tsx:413]: nitpick: Maybe we should rename this to `closeNode`

**Review by driskull — APPROVED** [review]
🎉 Looks good 🎉 
🌮

File: src/components/panel/panel.tsx [hunk 5]
```diff
@@ -196,7 +223,7 @@ export class Panel implements ConditionalSlotComponent, InteractiveComponent {
   @Event() calcitePanelDismiss: EventEmitter;
 
   /**
-   * Fires when the close button is clicked.
+   * Fires when there is a change in dismissed prop value .
```
Comment (benelan) [inline src/components/panel/panel.tsx:226]: Nitpick:
We try not to abbreviate words in the doc. Also property/slot/event/css-var names should be in back ticks. So:

> Fires when there is a change to the \`dismissed\` property value.

File: src/components/panel/panel.tsx (hunk 4 above)
Comment (anveshmekala) [reply src/components/panel/panel.tsx:413]: Do we have to change `setDismissRef` to `setCloseRef` and  `dismissButtonEl` to `closeButtonEl` . thoughts?

Comment (driskull) [reply src/components/panel/panel.tsx:413]: That would be nice too.

**Pull request merged**